SNOW-726880: Could not build wheels on MacOS

Please answer these questions before submitting your issue. Thanks!

1. What version of Python are you using?

```
Python 3.8.12 (default, Oct 12 2021, 06:23:56) 
[Clang 10.0.0 ]
```

2. What operating system and processor architecture are you using?

`macOS-10.16-x86_64-i386-64bit`

3. What are the component versions in the environment (`pip freeze`)?

```
build==0.10.0
certifi @ file:///private/var/folders/sy/f16zz6x50xz3113nwtb9bvq00000gp/T/abs_477u68wvzm/croot/certifi_1671487773341/work/certifi
packaging==23.0
pyproject_hooks==1.0.0
tomli==2.0.1
```

4. What did you do?

Tried to build:

```
╰─ git clone https://github.com/snowflakedb/snowflake-connector-python.git
╰─ cd snowflake-connector-python
╰─ brew install gcc --HEAD
```

```
╰─ gcc --version
Apple clang version 14.0.0 (clang-1400.0.29.202)
Target: x86_64-apple-darwin22.1.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

```
╰─ python -m build --wheel .

...
building 'snowflake.connector.arrow_iterator' extension
gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/berkay/miniconda3/envs/snowflake/include -arch x86_64 -I/Users/berkay/miniconda3/envs/snowflake/include -arch x86_64 -I/usr/local/opt/openssl/include -Isrc/snowflake/connector -Isrc/snowflake/connector/cpp/ArrowIterator -Isrc/snowflake/connector/cpp/Logging -I/private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/include -I/Users/berkay/miniconda3/envs/snowflake/include/python3.8 -c src/snowflake/connector/arrow_iterator.cpp -o build/temp.macosx-10.9-x86_64-cpython-38/src/snowflake/connector/arrow_iterator.o -isystem/private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include -isystem/private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/numpy/core/include -std=c++17 -D_GLIBCXX_USE_CXX11_ABI=0
In file included from src/snowflake/connector/arrow_iterator.cpp:842:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/api.h:22:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/array.h:37:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/array/array_base.h:26:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/array/data.h:28:
/private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/type.h:1745:36: error: 'get<arrow::FieldPath, arrow::FieldPath, std::string, std::vector<arrow::FieldRef>>' is unavailable: introduced in macOS 10.13
    if (IsFieldPath()) return std::get<FieldPath>(impl_).indices().size() > 1;
                                   ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<arrow::FieldPath, arrow::FieldPath, std::string, std::vector<arrow::FieldRef>>' has been explicitly marked unavailable here
constexpr const _Tp& get(const variant<_Types...>& __v) {
                     ^
In file included from src/snowflake/connector/arrow_iterator.cpp:842:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/api.h:22:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/array.h:37:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/array/array_base.h:26:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/array/data.h:28:
/private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/type.h:1750:34: error: 'get<arrow::FieldPath, arrow::FieldPath, std::string, std::vector<arrow::FieldRef>>' is unavailable: introduced in macOS 10.13
    return IsFieldPath() ? &std::get<FieldPath>(impl_) : NULLPTR;
                                 ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<arrow::FieldPath, arrow::FieldPath, std::string, std::vector<arrow::FieldRef>>' has been explicitly marked unavailable here
constexpr const _Tp& get(const variant<_Types...>& __v) {
                     ^
In file included from src/snowflake/connector/arrow_iterator.cpp:842:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/api.h:22:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/array.h:37:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/array/array_base.h:26:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/array/data.h:28:
/private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/type.h:1753:29: error: 'get<std::string, arrow::FieldPath, std::string, std::vector<arrow::FieldRef>>' is unavailable: introduced in macOS 10.13
    return IsName() ? &std::get<std::string>(impl_) : NULLPTR;
                            ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<std::string, arrow::FieldPath, std::string, std::vector<arrow::FieldRef>>' has been explicitly marked unavailable here
constexpr const _Tp& get(const variant<_Types...>& __v) {
                     ^
In file included from src/snowflake/connector/arrow_iterator.cpp:842:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/api.h:22:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/array.h:37:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/array/array_base.h:26:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/array/data.h:28:
/private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/type.h:1757:24: error: 'get<std::vector<arrow::FieldRef>, arrow::FieldPath, std::string, std::vector<arrow::FieldRef>>' is unavailable: introduced in macOS 10.13
               ? &std::get<std::vector<FieldRef>>(impl_)
                       ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<std::vector<arrow::FieldRef>, arrow::FieldPath, std::string, std::vector<arrow::FieldRef>>' has been explicitly marked unavailable here
constexpr const _Tp& get(const variant<_Types...>& __v) {
                     ^
In file included from src/snowflake/connector/arrow_iterator.cpp:842:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/api.h:29:
/private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/datum.h:139:17: error: 'get<std::shared_ptr<arrow::ArrayData>, arrow::Datum::Empty, std::shared_ptr<arrow::Scalar>, std::shared_ptr<arrow::ArrayData>, std::shared_ptr<arrow::ChunkedArray>, std::shared_ptr<arrow::RecordBatch>, std::shared_ptr<arrow::Table>>' is unavailable: introduced in macOS 10.13
    return std::get<std::shared_ptr<ArrayData>>(this->value);
                ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<std::shared_ptr<arrow::ArrayData>, arrow::Datum::Empty, std::shared_ptr<arrow::Scalar>, std::shared_ptr<arrow::ArrayData>, std::shared_ptr<arrow::ChunkedArray>, std::shared_ptr<arrow::RecordBatch>, std::shared_ptr<arrow::Table>>' has been explicitly marked unavailable here
constexpr const _Tp& get(const variant<_Types...>& __v) {
                     ^
In file included from src/snowflake/connector/arrow_iterator.cpp:842:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/api.h:29:
/private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/datum.h:152:17: error: 'get<std::shared_ptr<arrow::ChunkedArray>, arrow::Datum::Empty, std::shared_ptr<arrow::Scalar>, std::shared_ptr<arrow::ArrayData>, std::shared_ptr<arrow::ChunkedArray>, std::shared_ptr<arrow::RecordBatch>, std::shared_ptr<arrow::Table>>' is unavailable: introduced in macOS 10.13
    return std::get<std::shared_ptr<ChunkedArray>>(this->value);
                ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<std::shared_ptr<arrow::ChunkedArray>, arrow::Datum::Empty, std::shared_ptr<arrow::Scalar>, std::shared_ptr<arrow::ArrayData>, std::shared_ptr<arrow::ChunkedArray>, std::shared_ptr<arrow::RecordBatch>, std::shared_ptr<arrow::Table>>' has been explicitly marked unavailable here
constexpr const _Tp& get(const variant<_Types...>& __v) {
                     ^
In file included from src/snowflake/connector/arrow_iterator.cpp:842:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/api.h:29:
/private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/datum.h:156:17: error: 'get<std::shared_ptr<arrow::RecordBatch>, arrow::Datum::Empty, std::shared_ptr<arrow::Scalar>, std::shared_ptr<arrow::ArrayData>, std::shared_ptr<arrow::ChunkedArray>, std::shared_ptr<arrow::RecordBatch>, std::shared_ptr<arrow::Table>>' is unavailable: introduced in macOS 10.13
    return std::get<std::shared_ptr<RecordBatch>>(this->value);
                ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<std::shared_ptr<arrow::RecordBatch>, arrow::Datum::Empty, std::shared_ptr<arrow::Scalar>, std::shared_ptr<arrow::ArrayData>, std::shared_ptr<arrow::ChunkedArray>, std::shared_ptr<arrow::RecordBatch>, std::shared_ptr<arrow::Table>>' has been explicitly marked unavailable here
constexpr const _Tp& get(const variant<_Types...>& __v) {
                     ^
In file included from src/snowflake/connector/arrow_iterator.cpp:842:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/api.h:29:
/private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/datum.h:160:17: error: 'get<std::shared_ptr<arrow::Table>, arrow::Datum::Empty, std::shared_ptr<arrow::Scalar>, std::shared_ptr<arrow::ArrayData>, std::shared_ptr<arrow::ChunkedArray>, std::shared_ptr<arrow::RecordBatch>, std::shared_ptr<arrow::Table>>' is unavailable: introduced in macOS 10.13
    return std::get<std::shared_ptr<Table>>(this->value);
                ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<std::shared_ptr<arrow::Table>, arrow::Datum::Empty, std::shared_ptr<arrow::Scalar>, std::shared_ptr<arrow::ArrayData>, std::shared_ptr<arrow::ChunkedArray>, std::shared_ptr<arrow::RecordBatch>, std::shared_ptr<arrow::Table>>' has been explicitly marked unavailable here
constexpr const _Tp& get(const variant<_Types...>& __v) {
                     ^
In file included from src/snowflake/connector/arrow_iterator.cpp:842:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/api.h:29:
/private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/datum.h:164:17: error: 'get<std::shared_ptr<arrow::Scalar>, arrow::Datum::Empty, std::shared_ptr<arrow::Scalar>, std::shared_ptr<arrow::ArrayData>, std::shared_ptr<arrow::ChunkedArray>, std::shared_ptr<arrow::RecordBatch>, std::shared_ptr<arrow::Table>>' is unavailable: introduced in macOS 10.13
    return std::get<std::shared_ptr<Scalar>>(this->value);
                ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<std::shared_ptr<arrow::Scalar>, arrow::Datum::Empty, std::shared_ptr<arrow::Scalar>, std::shared_ptr<arrow::ArrayData>, std::shared_ptr<arrow::ChunkedArray>, std::shared_ptr<arrow::RecordBatch>, std::shared_ptr<arrow::Table>>' has been explicitly marked unavailable here
constexpr const _Tp& get(const variant<_Types...>& __v) {
                     ^
In file included from src/snowflake/connector/arrow_iterator.cpp:853:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/util/value_parsing.h:38:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/vendored/datetime.h:20:
/private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/vendored/datetime/date.h:1069:22: error: 'uncaught_exceptions' is unavailable: introduced in macOS 10.12
                std::uncaught_exceptions() == 0 &&
                     ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/exception:134:63: note: 'uncaught_exceptions' has been explicitly marked unavailable here
_LIBCPP_FUNC_VIS _LIBCPP_AVAILABILITY_UNCAUGHT_EXCEPTIONS int uncaught_exceptions() _NOEXCEPT;
                                                              ^
In file included from src/snowflake/connector/arrow_iterator.cpp:858:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/compute/api.h:27:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/compute/api_aggregate.h:23:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/compute/function.h:27:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/compute/kernel.h:32:
/private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/compute/exec.h:333:41: error: 'get<arrow::ArraySpan, arrow::ArraySpan, std::shared_ptr<arrow::ArrayData>>' is unavailable: introduced in macOS 10.13
    return const_cast<ArraySpan*>(&std::get<ArraySpan>(this->value));
                                        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<arrow::ArraySpan, arrow::ArraySpan, std::shared_ptr<arrow::ArrayData>>' has been explicitly marked unavailable here
constexpr const _Tp& get(const variant<_Types...>& __v) {
                     ^
In file included from src/snowflake/connector/arrow_iterator.cpp:858:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/compute/api.h:27:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/compute/api_aggregate.h:23:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/compute/function.h:27:
In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/compute/kernel.h:32:
/private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/build-env-fyhm8z5b/lib/python3.8/site-packages/pyarrow/include/arrow/compute/exec.h:338:17: error: 'get<std::shared_ptr<arrow::ArrayData>, arrow::ArraySpan, std::shared_ptr<arrow::ArrayData>>' is unavailable: introduced in macOS 10.13
    return std::get<std::shared_ptr<ArrayData>>(this->value);
                ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<std::shared_ptr<arrow::ArrayData>, arrow::ArraySpan, std::shared_ptr<arrow::ArrayData>>' has been explicitly marked unavailable here
constexpr const _Tp& get(const variant<_Types...>& __v) {
                     ^
src/snowflake/connector/arrow_iterator.cpp:5819:3: warning: 'tp_print' is deprecated [-Wdeprecated-declarations]
  0, /*tp_print*/
  ^
/Users/berkay/miniconda3/envs/snowflake/include/python3.8/cpython/object.h:260:5: note: 'tp_print' has been explicitly marked deprecated here
    Py_DEPRECATED(3.8) int (*tp_print)(PyObject *, FILE *, int);
    ^
/Users/berkay/miniconda3/envs/snowflake/include/python3.8/pyport.h:515:54: note: expanded from macro 'Py_DEPRECATED'
#define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                     ^
src/snowflake/connector/arrow_iterator.cpp:5998:3: warning: 'tp_print' is deprecated [-Wdeprecated-declarations]
  0, /*tp_print*/
  ^
/Users/berkay/miniconda3/envs/snowflake/include/python3.8/cpython/object.h:260:5: note: 'tp_print' has been explicitly marked deprecated here
    Py_DEPRECATED(3.8) int (*tp_print)(PyObject *, FILE *, int);
    ^
/Users/berkay/miniconda3/envs/snowflake/include/python3.8/pyport.h:515:54: note: expanded from macro 'Py_DEPRECATED'
#define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                     ^
2 warnings and 12 errors generated.
error: command '/usr/bin/gcc' failed with exit code 1

ERROR Backend subprocess exited when trying to invoke build_wheel
```

Similar issues when running `tox`:

```
Building wheels for collected packages: snowflake-connector-python
  Building wheel for snowflake-connector-python (pyproject.toml): started
  Building wheel for snowflake-connector-python (pyproject.toml): finished with status 'error'
  error: subprocess-exited-with-error
  
  × Building wheel for snowflake-connector-python (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [457 lines of output]

...

/lib/python3.8/site-packages/pyarrow/include/arrow/array/data.h:28:
      /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/type.h:1745:36: error: 'get<arrow::FieldPath, arrow::FieldPath, std::string, std::vector<arrow::FieldRef>>' is unavailable: introduced in macOS 10.13
          if (IsFieldPath()) return std::get<FieldPath>(impl_).indices().size() > 1;
                                         ^
      /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<arrow::FieldPath, arrow::FieldPath, std::string, std::vector<arrow::FieldRef>>' has been explicitly marked unavailable here
      constexpr const _Tp& get(const variant<_Types...>& __v) {
                           ^
      In file included from src/snowflake/connector/arrow_iterator.cpp:842:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/api.h:22:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/array.h:37:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/array/array_base.h:26:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/array/data.h:28:
      /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/type.h:1750:34: error: 'get<arrow::FieldPath, arrow::FieldPath, std::string, std::vector<arrow::FieldRef>>' is unavailable: introduced in macOS 10.13
          return IsFieldPath() ? &std::get<FieldPath>(impl_) : NULLPTR;
                                       ^
      /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<arrow::FieldPath, arrow::FieldPath, std::string, std::vector<arrow::FieldRef>>' has been explicitly marked unavailable here
      constexpr const _Tp& get(const variant<_Types...>& __v) {
                           ^
      In file included from src/snowflake/connector/arrow_iterator.cpp:842:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/api.h:22:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/array.h:37:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/array/array_base.h:26:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/array/data.h:28:
      /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/type.h:1753:29: error: 'get<std::string, arrow::FieldPath, std::string, std::vector<arrow::FieldRef>>' is unavailable: introduced in macOS 10.13
          return IsName() ? &std::get<std::string>(impl_) : NULLPTR;
                                  ^
      /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<std::string, arrow::FieldPath, std::string, std::vector<arrow::FieldRef>>' has been explicitly marked unavailable here
      constexpr const _Tp& get(const variant<_Types...>& __v) {
                           ^
      In file included from src/snowflake/connector/arrow_iterator.cpp:842:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/api.h:22:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/array.h:37:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/array/array_base.h:26:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/array/data.h:28:
      /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/type.h:1757:24: error: 'get<std::vector<arrow::FieldRef>, arrow::FieldPath, std::string, std::vector<arrow::FieldRef>>' is unavailable: introduced in macOS 10.13
                     ? &std::get<std::vector<FieldRef>>(impl_)
                             ^
      /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<std::vector<arrow::FieldRef>, arrow::FieldPath, std::string, std::vector<arrow::FieldRef>>' has been explicitly marked unavailable here
      constexpr const _Tp& get(const variant<_Types...>& __v) {
                           ^
      In file included from src/snowflake/connector/arrow_iterator.cpp:842:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/api.h:29:
      /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/datum.h:139:17: error: 'get<std::shared_ptr<arrow::ArrayData>, arrow::Datum::Empty, std::shared_ptr<arrow::Scalar>, std::shared_ptr<arrow::ArrayData>, std::shared_ptr<arrow::ChunkedArray>, std::shared_ptr<arrow::RecordBatch>, std::shared_ptr<arrow::Table>>' is unavailable: introduced in macOS 10.13
          return std::get<std::shared_ptr<ArrayData>>(this->value);
                      ^
      /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<std::shared_ptr<arrow::ArrayData>, arrow::Datum::Empty, std::shared_ptr<arrow::Scalar>, std::shared_ptr<arrow::ArrayData>, std::shared_ptr<arrow::ChunkedArray>, std::shared_ptr<arrow::RecordBatch>, std::shared_ptr<arrow::Table>>' has been explicitly marked unavailable here
      constexpr const _Tp& get(const variant<_Types...>& __v) {
                           ^
      In file included from src/snowflake/connector/arrow_iterator.cpp:842:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/api.h:29:
      /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/datum.h:152:17: error: 'get<std::shared_ptr<arrow::ChunkedArray>, arrow::Datum::Empty, std::shared_ptr<arrow::Scalar>, std::shared_ptr<arrow::ArrayData>, std::shared_ptr<arrow::ChunkedArray>, std::shared_ptr<arrow::RecordBatch>, std::shared_ptr<arrow::Table>>' is unavailable: introduced in macOS 10.13
          return std::get<std::shared_ptr<ChunkedArray>>(this->value);
                      ^
      /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<std::shared_ptr<arrow::ChunkedArray>, arrow::Datum::Empty, std::shared_ptr<arrow::Scalar>, std::shared_ptr<arrow::ArrayData>, std::shared_ptr<arrow::ChunkedArray>, std::shared_ptr<arrow::RecordBatch>, std::shared_ptr<arrow::Table>>' has been explicitly marked unavailable here
      constexpr const _Tp& get(const variant<_Types...>& __v) {
                           ^
      In file included from src/snowflake/connector/arrow_iterator.cpp:842:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/api.h:29:
      /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/datum.h:156:17: error: 'get<std::shared_ptr<arrow::RecordBatch>, arrow::Datum::Empty, std::shared_ptr<arrow::Scalar>, std::shared_ptr<arrow::ArrayData>, std::shared_ptr<arrow::ChunkedArray>, std::shared_ptr<arrow::RecordBatch>, std::shared_ptr<arrow::Table>>' is unavailable: introduced in macOS 10.13
          return std::get<std::shared_ptr<RecordBatch>>(this->value);
                      ^
      /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<std::shared_ptr<arrow::RecordBatch>, arrow::Datum::Empty, std::shared_ptr<arrow::Scalar>, std::shared_ptr<arrow::ArrayData>, std::shared_ptr<arrow::ChunkedArray>, std::shared_ptr<arrow::RecordBatch>, std::shared_ptr<arrow::Table>>' has been explicitly marked unavailable here
      constexpr const _Tp& get(const variant<_Types...>& __v) {
                           ^
      In file included from src/snowflake/connector/arrow_iterator.cpp:842:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/api.h:29:
      /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/datum.h:160:17: error: 'get<std::shared_ptr<arrow::Table>, arrow::Datum::Empty, std::shared_ptr<arrow::Scalar>, std::shared_ptr<arrow::ArrayData>, std::shared_ptr<arrow::ChunkedArray>, std::shared_ptr<arrow::RecordBatch>, std::shared_ptr<arrow::Table>>' is unavailable: introduced in macOS 10.13
          return std::get<std::shared_ptr<Table>>(this->value);
                      ^
      /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<std::shared_ptr<arrow::Table>, arrow::Datum::Empty, std::shared_ptr<arrow::Scalar>, std::shared_ptr<arrow::ArrayData>, std::shared_ptr<arrow::ChunkedArray>, std::shared_ptr<arrow::RecordBatch>, std::shared_ptr<arrow::Table>>' has been explicitly marked unavailable here
      constexpr const _Tp& get(const variant<_Types...>& __v) {
                           ^
      In file included from src/snowflake/connector/arrow_iterator.cpp:842:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/api.h:29:
      /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/datum.h:164:17: error: 'get<std::shared_ptr<arrow::Scalar>, arrow::Datum::Empty, std::shared_ptr<arrow::Scalar>, std::shared_ptr<arrow::ArrayData>, std::shared_ptr<arrow::ChunkedArray>, std::shared_ptr<arrow::RecordBatch>, std::shared_ptr<arrow::Table>>' is unavailable: introduced in macOS 10.13
          return std::get<std::shared_ptr<Scalar>>(this->value);
                      ^
      /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<std::shared_ptr<arrow::Scalar>, arrow::Datum::Empty, std::shared_ptr<arrow::Scalar>, std::shared_ptr<arrow::ArrayData>, std::shared_ptr<arrow::ChunkedArray>, std::shared_ptr<arrow::RecordBatch>, std::shared_ptr<arrow::Table>>' has been explicitly marked unavailable here
      constexpr const _Tp& get(const variant<_Types...>& __v) {
                           ^
      In file included from src/snowflake/connector/arrow_iterator.cpp:853:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/util/value_parsing.h:38:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/vendored/datetime.h:20:
      /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/vendored/datetime/date.h:1069:22: error: 'uncaught_exceptions' is unavailable: introduced in macOS 10.12
                      std::uncaught_exceptions() == 0 &&
                           ^
      /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/exception:134:63: note: 'uncaught_exceptions' has been explicitly marked unavailable here
      _LIBCPP_FUNC_VIS _LIBCPP_AVAILABILITY_UNCAUGHT_EXCEPTIONS int uncaught_exceptions() _NOEXCEPT;
                                                                    ^
      In file included from src/snowflake/connector/arrow_iterator.cpp:858:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/compute/api.h:27:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/compute/api_aggregate.h:23:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/compute/function.h:27:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/compute/kernel.h:32:
      /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/compute/exec.h:333:41: error: 'get<arrow::ArraySpan, arrow::ArraySpan, std::shared_ptr<arrow::ArrayData>>' is unavailable: introduced in macOS 10.13
          return const_cast<ArraySpan*>(&std::get<ArraySpan>(this->value));
                                              ^
      /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<arrow::ArraySpan, arrow::ArraySpan, std::shared_ptr<arrow::ArrayData>>' has been explicitly marked unavailable here
      constexpr const _Tp& get(const variant<_Types...>& __v) {
                           ^
      In file included from src/snowflake/connector/arrow_iterator.cpp:858:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/compute/api.h:27:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/compute/api_aggregate.h:23:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/compute/function.h:27:
      In file included from /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/compute/kernel.h:32:
      /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/pyarrow/include/arrow/compute/exec.h:338:17: error: 'get<std::shared_ptr<arrow::ArrayData>, arrow::ArraySpan, std::shared_ptr<arrow::ArrayData>>' is unavailable: introduced in macOS 10.13
          return std::get<std::shared_ptr<ArrayData>>(this->value);
                      ^
      /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/variant:1544:22: note: 'get<std::shared_ptr<arrow::ArrayData>, arrow::ArraySpan, std::shared_ptr<arrow::ArrayData>>' has been explicitly marked unavailable here
      constexpr const _Tp& get(const variant<_Types...>& __v) {
                           ^
      src/snowflake/connector/arrow_iterator.cpp:5819:3: warning: 'tp_print' is deprecated [-Wdeprecated-declarations]
        0, /*tp_print*/
        ^
      /Users/berkay/miniconda3/envs/pave/include/python3.8/cpython/object.h:260:5: note: 'tp_print' has been explicitly marked deprecated here
          Py_DEPRECATED(3.8) int (*tp_print)(PyObject *, FILE *, int);
          ^
      /Users/berkay/miniconda3/envs/pave/include/python3.8/pyport.h:515:54: note: expanded from macro 'Py_DEPRECATED'
      #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                           ^
      src/snowflake/connector/arrow_iterator.cpp:5998:3: warning: 'tp_print' is deprecated [-Wdeprecated-declarations]
        0, /*tp_print*/
        ^
      /Users/berkay/miniconda3/envs/pave/include/python3.8/cpython/object.h:260:5: note: 'tp_print' has been explicitly marked deprecated here
          Py_DEPRECATED(3.8) int (*tp_print)(PyObject *, FILE *, int);
          ^
      /Users/berkay/miniconda3/envs/pave/include/python3.8/pyport.h:515:54: note: expanded from macro 'Py_DEPRECATED'
      #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                           ^
      2 warnings and 12 errors generated.
      /private/var/folders/z2/yys4x2cs36gblb3t0n2j_ckh0000gn/T/pip-build-env-s31hl957/overlay/lib/python3.8/site-packages/setuptools/command/build_py.py:202: SetuptoolsDeprecationWarning:     Installing 'snowflake.connector.cpp.ArrowIterator' as data is deprecated, please list it in `packages`.
          !!

...

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for snowflake-connector-python
Failed to build snowflake-connector-python
ERROR: Could not build wheels for snowflake-connector-python, which is required to install pyproject.toml-based projects
```

6. What did you expect to see?

   Building the wheels without any errors

--- a/setup.py
+++ b/setup.py
@@ -158,6 +158,8 @@ if _ABLE_TO_COMPILE_EXTENSIONS:
                     if "std=" not in os.environ.get("CXXFLAGS", ""):
                         ext.extra_compile_args.append("-std=c++17")
                         ext.extra_compile_args.append("-D_GLIBCXX_USE_CXX11_ABI=0")
+                    if sys.platform == "darwin":
+                        ext.extra_compile_args.append("-mmacosx-version-min=10.13")
 
                 ext.library_dirs.append(
                     os.path.join(current_dir, self.build_lib, "snowflake", "connector")

